### PR TITLE
fix(map-overlay): extent not moving when intersects with the screen at least 10%

### LINKED ISSUE #729: Map extent is changed when loading new overlay
this dosent happen all the time, but i managed create a reproduction:
1. open: https://goo.gl/KammwB
2. go back in time 2 steps (left keyboard key, left key above timeline)
3. map extent is changed

### PR DESCRIPTION
**Pull request by AmitMY**
resolve #729 

@idanbarak 10% seems enough.
Do you want a demo, or just merge it and you'll see it next version?

### CODE REVIEW

File: src/app/app-reducers/effects/map.app.effects.spec.ts [hunk 1]
```diff
@@ -491,48 +491,13 @@ describe('MapAppEffects', () => {
...
-	// 		expect(communicator.loadInitialMapSource).toHaveBeenCalled();
-	// 	});
-	// });
-
-	describe('onDisplayOverlay$ communicator should set Layer on map, by isExtentContainedInPolygon', () => {
+	describe('onDisplayOverlay$ communicator should set Layer on map, by getExtentIntersectionRatioInPolygon', () => {
 		const fake_layer = {};
 		const fake_extent = [1, 2, 3, 4];
-		let fakeCommuincator: CommunicatorEntity;
+		let fakeCommunicator: CommunicatorEntity;
```
Comment (osadan) [inline src/app/app-reducers/effects/map.app.effects.spec.ts]: good jbo

File: src/app/app-reducers/effects/map.app.effects.ts [hunk 2]
```diff
@@ -208,11 +209,10 @@ export class MapAppEffects {
 		})
 		.filter(([overlay]: [Overlay]) => !isEmpty(overlay) && overlay.isFullOverlay)
 		.flatMap(([overlay, map_id, position]: [Overlay, string, Position]) => {
-
-			const isInside = isExtentContainedInPolygon(position.boundingBox, overlay.footprint);
+			const intersection = getExtentIntersectionRatioInPolygon(position.boundingBox, overlay.footprint);
 
 			let extent;
-			if (isInside) {
+			if (intersection > 0.1) {
```
Comment (osadan) [inline src/app/app-reducers/effects/map.app.effects.ts]: config

File: src/app/packages/core/utils/calc-extent.ts [hunk 3]
```diff
@@ -51,3 +55,20 @@ export function isExtentContainedInPolygon(extent: GeoJSON.Point[], footprint: G
 	const centerPoint = center(extentPoly);
 	return inside(centerPoint, footprintFeature);
 }
+
+export function getExtentIntersectionRatioInPolygon(extent: GeoJSON.Point[], footprint: GeoJSON.MultiPolygon): number {
```
Comment (osadan) [inline src/app/packages/core/utils/calc-extent.ts]: very nice

**Comment by AmitMY** [comment]
@osadan It is done.

File: src/app/app-reducers/effects/map.app.effects.ts [hunk 4]
```diff
@@ -209,10 +210,10 @@ export class MapAppEffects {
 		})
 		.filter(([overlay]: [Overlay]) => !isEmpty(overlay) && overlay.isFullOverlay)
 		.flatMap(([overlay, map_id, position]: [Overlay, string, Position]) => {
-			const intersection = getExtentIntersectionRatioInPolygon(position.boundingBox, overlay.footprint);
+			const intersection = getFootprintIntersectionRatioInExtent(position.boundingBox, overlay.footprint);
 
 			let extent;
-			if (intersection > 0.1) {
+			if (intersection > this.config.overlayCoverage) {
```
Comment (osadan) [inline src/app/app-reducers/effects/map.app.effects.ts]: nice

**Pull request merged**